### DISCUSSION TITLE
Build Error compiling APK

### DISCUSSION THREAD

**Discussion by alexbrowning-nz** [781flyingdutchman/background_downloader#316] [Q&A]
Hi All,

First off; I'm a relatively new Android developer, so please bear with me.

I'm trying to compile an application that uses the background_downloader and I get the following error when doing so :-

Execution failed for task ':background_downloader:compileReleaseKotlin'.
> Error while evaluating property 'filteredArgumentsMap' of task ':background_downloader:compileReleaseKotlin'.
   > Could not resolve all files for configuration ':background_downloader:kotlinCompilerPluginClasspathRelease'.
      > Could not find org.jetbrains.kotlin:kotlin-serialization-compiler-plugin-embeddable:1.7.10.
      
The output goes on to say that it tried looking in a few locations :-

https://dl.google.com/dl/android/maven2/org/jetbrains/kotlin/kotlin-serialization-compiler-plugin-embeddable/1.7.10/kotlin-serialization-compiler-plugin-embeddable-1.7.10.pom

https://repo.maven.apache.org/maven2/org/jetbrains/kotlin/kotlin-serialization-compiler-plugin-embeddable/1.7.10/kotlin-serialization-compiler-plugin-embeddable-1.7.10.pom

And notably :-

https://plugins.gradle.org/m2/org/jetbrains/kotlin/kotlin-serialization-compiler-plugin-embeddable/1.7.10/kotlin-serialization-compiler-plugin-embeddable-1.7.10.pom

where the [kotlin-serialization/](https://plugins.gradle.org/m2/org/jetbrains/kotlin/kotlin-serialization/) seems to exist, but the embeddable does not

The repos I'm using are :-

google()
mavenCentral()
maven {
            url = uri("https://plugins.gradle.org/m2/")
        }
        
Which I've tried in buildscript and allprojects in my build.gradle file

Flutter doctor outputs all ticks for Flutter, Android Toolchain and Studio

Now, I know I'm probably doing something a little dumb, but I have done some searching and not found much I can apply, so any pointers would be gratefully received.

**Comment by 781flyingdutchman** [comment]
You need to change your Kotlin version to 1.9

**Comment by alexbrowning-nz** [comment]
Hi,

Thanks for your reply.

I was using 1.9.23, so I selected 1.9.0 and also 1.9.22 and tried to recompile, but I got the same errors.

I managed to get it compiling by updating my IDE, it was seemingly locked to 1.8 even though it allowed me to select 1.9.X as it's target Kotlin version.

Including this reply in case it helps anyone else or jogs anyone else's brain.

Thanks,

Alex.
